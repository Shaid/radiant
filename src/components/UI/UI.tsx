import React from 'react'
import styled, { breakpoints } from 'theme'

import EventsPanel from 'components/EventsPanel'
import Room from 'components/Room'
import StatusPanel from 'components/StatusPanel'

import { IProps, IDefaultProps, IState } from './UI.d'

const UI = styled.section`
  display: flex;
  flex-flow: column nowrap;
  height: 100%;
  position: relative;
  padding: 1rem;

  @media ${breakpoints.desktop} {
    padding: 1.5rem;
  }
`

export default class extends React.PureComponent<IProps, IState> { // eslint-disable-line react/prefer-stateless-function
  static defaultProps: Partial<IDefaultProps> = {}

  render() { // eslint-disable-line class-methods-use-this
    // const {} = this.props as PropsWithDefaults
    return (
      <UI>
        <Room />
        <EventsPanel />
        <StatusPanel />
      </UI>
    )
  }
}
