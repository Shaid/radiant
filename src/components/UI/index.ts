export { default } from './UI'

// NOTE: uncomment this line to pass through named exports
// export * from './UI'
