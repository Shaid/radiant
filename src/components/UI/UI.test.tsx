import React from 'react'
import { shallow } from 'enzyme'

import UI from '.'

describe('<UI />', () => {
  it('renders', () => {
    const wrapper = shallow(<UI />)
    expect(wrapper).toMatchSnapshot()
  })

  // @TODO: tests for UI
})
