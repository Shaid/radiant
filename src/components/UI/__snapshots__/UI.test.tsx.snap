// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<UI /> renders 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <_class />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": Array [
        <_class />,
        <_class />,
        <_default />,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": Array [],
      "componentStyle": ComponentStyle {
        "componentId": "sc-dxgOiQ",
        "isStatic": true,
        "rules": Array [
          "
  display: flex;
  flex-flow: column nowrap;
  height: 100%;
  position: relative;
  padding: 1rem;

  @media ",
          "all and (orientation: landscape)",
          " {
    padding: 0 1.5rem;
  }
",
        ],
      },
      "displayName": "styled.section",
      "foldedComponentIds": Array [],
      "render": [Function],
      "styledComponentId": "sc-dxgOiQ",
      "target": "section",
      "toString": [Function],
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
    },
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <_class />,
          <_class />,
          <_default />,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": Object {
        "$$typeof": Symbol(react.forward_ref),
        "attrs": Array [],
        "componentStyle": ComponentStyle {
          "componentId": "sc-dxgOiQ",
          "isStatic": true,
          "rules": Array [
            "
  display: flex;
  flex-flow: column nowrap;
  height: 100%;
  position: relative;
  padding: 1rem;

  @media ",
            "all and (orientation: landscape)",
            " {
    padding: 0 1.5rem;
  }
",
          ],
        },
        "displayName": "styled.section",
        "foldedComponentIds": Array [],
        "render": [Function],
        "styledComponentId": "sc-dxgOiQ",
        "target": "section",
        "toString": [Function],
        "warnTooManyClasses": [Function],
        "withComponent": [Function],
      },
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
  Symbol(enzyme.__childContext__): null,
}
`;
